engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -918,6 +924,12 @@ namespace dmGameSystem
                     component->m_VertSize = vert_size;
                     component->m_VertexDeclaration = new_vert_decl;
                 }
+
+                if (dmRender::GetMaterialVertexSpace(GetMaterial(component, component->m_Resource)) == dmRenderDDF::MaterialDesc::VERTEX_SPACE_LOCAL)
+                {
+                    CreateVertexBuffer(world, br, GetVertexSize(component)); // Will inc ref the buffer
+                    DecRefVertexBuffer(world, prev_buffer_resource->m_NameHash);
```

Comment:
We previously tried to increase the ref count for the buffer, but we hadn't allocated it yet!

Also, we need the vertex size to allocate it, so it has to be done after the vertex declaration is done.